external_parser/event_processors/joined_event.h

```diff
@@ -324,6 +326,77 @@ struct ccb_joined_event : public typed_joined_event {
   }
 };
 
+struct slates_joined_event : public typed_joined_event {
+  MultiSlotInteraction multi_slot_interaction;
+  float reward;
+  float original_reward;
+
+  ~slates_joined_event() = default;
+
+  bool is_skip_learn() const override { return multi_slot_interaction.skip_learn; }
+  void set_skip_learn(bool sl) override { multi_slot_interaction.skip_learn = sl; }
+
+  void set_apprentice_reward() override {}
+
+  bool fill_in_label(v_array<example *> &examples) const override {
+    size_t slot_index = 0;
+
+    for (auto *ex : examples) {
+      if (ex->l.slates.type ==  VW::slates::example_type::slot) {
+        auto &slot_label = ex->l.slates;
+        if (multi_slot_interaction.interaction_data.size() > slot_index) {
+          const auto &slot_data = multi_slot_interaction.interaction_data[slot_index];
+          if ((slot_data.actions.size() != 0) &&
+              (slot_data.probabilities.size() != 0)) {
+            if (slot_data.actions.size() != slot_data.probabilities.size()) {
+              VW::io::logger::log_warn(
+                  "actions and probabilities for event [{}] don't have the "
+                  "same size. Actions [{}], probabilities [{}]",
+                  slot_data.eventId, slot_data.actions.size(),
+                  slot_data.probabilities.size());
+              continue;
+            }
+
+            for (size_t i = 0; i < slot_data.actions.size(); i++) {
+              slot_label.probabilities.push_back(
+                {slot_data.actions[i], slot_data.probabilities[i]}
+              );
+            }
+          }
+        }
+        // process next slot from interaction_data vector
+        slot_index++;
+      }
+    }
+    return true;
+  }
+
+  void set_cost(v_array<example *> &examples, float reward,
+                size_t slot_offset = 0) const override {
+    if (examples[0]->l.slates.type ==  VW::slates::example_type::shared) {
```

Comment:
so if it isn't shared that means we won't set a cost but we will have set the probabilities and actions?

Reply:
I think setting this to `examples[0]` without checking if it is `shared` is ok based on [here](https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/vowpalwabbit/parse_slates_example_json.h#L158) and [here](https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/vowpalwabbit/parse_slates_example_json.h#L246)